contracts/base/ATM/ATMToken.sol

```diff
@@ -0,0 +1,285 @@
+    uint256 private _cap;
+    uint256 private _maxVestingsPerWallet;
+    address private _owner;
+
+    /* Structs */
+    struct VestingTokens {
+        address account;
+        uint256 amount;
+        uint256 start;
+        uint256 cliff;
+        uint256 deadline;
+    }
+
+    /* Mappings */
+    mapping (address => mapping(uint256 => VestingTokens)) private _vestingBalances;
+    mapping (address => uint256) public vestingsCount;
+    mapping (address => uint256) public assignedTokens;
+
+    /* Constructor */
+    constructor (
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals,
+        uint256 cap,
+        uint256 maxVestingsPerWallet
+    ) 
+        ERC20Detailed(_name, _symbol, _decimals)
+        public
+    {
+        _cap = cap;
+        _maxVestingsPerWallet = maxVestingsPerWallet;
+        _owner = msg.sender;
+    }
+
+    /* Functions */
+    /**
+     * @notice Returns the cap on the token's total supply
+     * @return The supply capped amount
+     */
+    function cap() 
+        public
+        view
+        returns (uint256) 
+    {
+        return _cap;
+    }
+
+    /**
+     * @notice Sets a new cap on the token's total supply.
+     * @param newCap The new capped amount of tokens
+     */
+    function setCap(
+        uint256 newCap
+    ) 
+        public
+        onlyOwner()
+    {
+        _cap = newCap;
+        emit NewCap(_cap);
+    }
+
+    /**
+     * @notice Increase account supply of specified token amount
+     * @param account The account to mint tokens for
+     * @param amount The amount of tokens to mint
+     * @return true if successful
+     */
+    function mint(
+        address account,
+        uint256 amount
+    ) 
+        public
+        onlyOwner()
+        returns (bool)
+    {
+        require(account != address(0), "MINT_TO_ZERO_ADDRESS");
+        _beforeTokenTransfer(address(0), account, amount);
+        _mint(account, amount);
+        return true;
+    }
+
+    /** @notice Creates `amount` tokens and assigns them to `account`, increasing
+     * the total supply.
+     *
+     * Includes a vesting period before address is allowed to use tokens
+     *
+     * Emits a {Transfer} event with `from` set to the zero address.
+     *
+     * Requirements
+     *
+     * - `to` cannot be the zero address.
+     *
+     * @param account The account which tokens will be assigned to
+     * @param amount The amount of tokens to be assigned
+     * @param cliff The length of time (in seconds) after which the tokens will start vesting
+     * @param vestingTime The length of the vesting period (in seconds)
+     */
+    function mintVesting(
+        address account,
+        uint256 amount,
+        uint256 cliff,
+        uint256 vestingTime
+    ) 
+        public
+        onlyOwner()
+    {
+        require(account != address(0), "MINT_TO_ZERO_ADDRESS");
+        require(vestingsCount[account] < _maxVestingsPerWallet, "MAX_VESTINGS_REACHED!");
```

Comment:
Probably we should remove ! from require message for compatibility with other messages.

Reply:
Removed